codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/CodecRegistrationCheckerCompilerPlugin.scala

```diff
@@ -30,11 +30,11 @@ class CodecRegistrationCheckerCompilerPlugin(override val global: Global) extend
     options.filterNot(_.startsWith("-")).headOption match {
       case Some(path) =>
         try {
-          val f = new File(path + File.separator + cacheFileName)
-          f.getCanonicalPath
-          pluginOptions.cacheFile = f
-          pluginOptions.oldTypes = {
-            val raf = new RandomAccessFile(f, "rw")
+          val cacheFile = new File(path + File.separator + directClassDescendantsCacheFileName)
+          cacheFile.getCanonicalPath
```

Comment:
Hmm looks like a dangling expression with no side effect 🤔 would be nice checking if it's a _thinko_ or some non-obvious hack

Reply:
Yes, I also was wondering about this line, but I checked that `getCanonicalPath` invoked other methods that can throw IOException e.g. when the `Path is not valid etc. I am not sure if this is **really** needed, but as I am not sure and this thing is doing some checks under the hood - I decided to leave it as it is.